### ISSUE TITLE
dart2js: Type from signature emitted in main unit when superclass in deferred unit

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#34219]
In the following example, `SystemMessage` is placed in the main unit, but the superclass `GeneratedMessage` is placed in a 'part' file.
It appears that the problem is that `DeferredLoadTask` does not walk the signature of the 'StaticUse' of the `KLocalFunction` to add the types in the signature to the dependencies.

main:
```dart
import 'bug1a7m1_lib1.dart' deferred as lib1;
import 'bug1a7m1_lib3.dart' deferred as lib3;

main() async {
  await lib1.loadLibrary();
  lib1.test1();
  await lib3.loadLibrary();
  lib3.test3();
}
```
bug1a7m1_lib1.dart:
```dart
import 'bug1a7m1_lib2.dart';

class SystemMessage extends GeneratedMessage {}

test1() {
  List<SystemMessage> xx = <Null>[];
  new GeneratedMessage();
// This closure has inferred type `int Function(SystemMessage, SystemMessage)`:
  xx.sort((a, b) => a == null ? (b == null ? 0 : 1) : (b == null ? -1 : 0));
}
```
bug1a7m1_lib2.dart:
```dart
class GeneratedMessage {}
```
bug1a7m1_lib3.dart:
```dart
import 'bug1a7m1_lib2.dart';

test3() {
  new GeneratedMessage();
}
```

**Comment by rakudrama** [comment]
A similar problem can be provoked with the bounds of generic function type parameters.

**Comment by rakudrama** [comment]
Fix available

**Comment by rakudrama** [comment]
Fixed via e1df47b15e798d73bf3049f55320ad7867dc2477
